modules/styling/stories/Stencils.stories.mdx

```diff
@@ -0,0 +1,270 @@
+import {SymbolDoc} from '@workday/canvas-kit-docs';
+
+import {CreateStencil} from './examples/CreateStencil';
+
+<Meta title="Features/Styling/Stencils" />
+
+## Stencil
+
+Stencils are a reusable function that returns `style` and `className` props in an object. A Stencil
+should apply to a single element. If your component has nested elements, you can either nest styles
+in the stencil or create a stencil for each nested element. If your component is a compound
+component, a stencil should be created for each subcomponet. If your component is a config
+component, a stencil can have nested styles.
+
+<SymbolDoc name="createStencil" />
+
+## Concepts
+
+### Base styles
+
+Base styles are always applied to a Stencil. All your default styles should go here. Base styles
+support psuedo selectors like `:focus-visible` or `:hover` as well as child selectors. Any selector
+supported by `@emotion/css` is valid here. All styles must be static and statically analyzable by
+the tranformer. If you need dynamic styling, look at Variables and Modifiers.
+
+### Variables
+
+Variables allow some properties to be dynamic. They work by creating CSS variables and are applied
+using the [style](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/style) property of an
+element to locally scope an override.
+
+Here's a simplified example:
+
+```tsx
+const myStencil = createStencil({
+  vars: {
+    defaultColor: 'red' // default value
+  },
+  base: ({defaultColor}) => {
+    color: defaultColor // `defaultColor` is '--abc123-defaultColor', not 'red'
+  }
+})
+
+const elemProps = myStencil({color: 'blue'}) // {style: {'--abc123-defaultColor': 'blue'}}
+
+<div {...elemProps} />
+```
+
+This will produce the following HTML:
+
+```html
+<style>
+  .css-abc123 {
+    --abc123-defaultColor: red;
+    color: var(--abc123-defaultColor);
+  }
+</style>
+<div class="css-123abc" style="--abc123-defaultColor: blue;"></div>
+```
```

Comment:
are you showing how `red` get override here?. The `this will produce the following html` is for the above code block, but it looks like below you're showing that applying a different value to the variable in the style attribute overrides it?